src/gui/guiiconprovider.cpp

```diff
@@ -144,5 +88,9 @@ void GuiIconProvider::configure()
 {
 #if (defined(Q_OS_UNIX) && !defined(Q_OS_MAC))
     m_useSystemTheme = Preferences::instance()->useSystemIconTheme();
+    if (!m_useSystemTheme)
```

Comment:
Maybe this??
```c++
#if (defined(Q_OS_UNIX) && !defined(Q_OS_MAC))
    m_useSystemTheme = Preferences::instance()->useSystemIconTheme();
    if (!m_useSystemTheme)
#endif
        QIcon::setThemeName("qbt-theme");
```

Reply:
Done.

Reply:
@sledgehammer999, @zeule: 

```cpp
if (!m_useSystemTheme)
#endif
``` 
Unless I am mistaken, the `if` line does nothing here.

Reply:
@sledgehammer999, @zeule: 

```cpp
if (!m_useSystemTheme)
#endif
``` 
Unless I am mistaken, the `if` line does nothing here.  

Can you please remove it?

Reply:
@LordNyriox it does if you're on linux/BSD :)

Reply:
@thalieht, @zeule:  What I mean is thst, currently, this `if` clause is immediately followed by an `endif`—with no code in between.  

You have to look at the full diff to see it.

Reply:
@thalieht, @zeule:  What I mean is thst, currently, this `if` clause is immediately followed by an `endif` line—with no code between those two lines.  

You have to look at the [full diff](https://github.com/qbittorrent/qBittorrent/pull/9295/files) to see it.

Reply:
Doesn't matter. `#if #endif` are preprocessor macros that based on the conditional, define whether the code inside them exists for the compiler or not. 
So if you're on linux you can read this as 
```
m_useSystemTheme = Preferences::instance()->useSystemIconTheme();
if (!m_useSystemTheme)
    QIcon::setThemeName("qbt-theme");
```

Reply:
@thalieht:  That is how it was before, yes. 

But in the "`fixup!` commit, the `QIcon::setThemeName("qbt-theme");` line was **moved** to *after* the `#endif`—leaving the `if (!m_useSystemTheme)` clause empty. 

In other words, there is **no code** inside this particular `if  #endif` macro right now.

Reply:
@LordNyriox 
You might want to learn about C++ preprocessor, these tutorials seems nice:
http://www.cplusplus.com/doc/tutorial/preprocessor/
https://en.cppreference.com/w/cpp/preprocessor/conditional

---
Or this? Duplicates but it avoids crossing `if` body.
```c++
#if (defined(Q_OS_UNIX) && !defined(Q_OS_MAC))
    m_useSystemTheme = Preferences::instance()->useSystemIconTheme();
    if (!m_useSystemTheme)
        QIcon::setThemeName("qbt-theme");
#else
    QIcon::setThemeName("qbt-theme");
#endif
```

Reply:
@Chocobo1 that's the way it was done before the "fixip!" commit. It's just @LordNyriox that doesn't get it.

Reply:
@sledgehammer999:  My point is that the *second* `if` statement in the block is no longer used, since there is **no code inside** it after the `fixup!` commit.

Reply:
@thalieht, @Chocobo1, @sledgehammer999:  

Keep in mind that I am limited to using a mobile browser right now, and can therefore only (easily) see the diff.  

Old Code: 
```cpp
#if (defined(Q_OS_UNIX) && !defined(Q_OS_MAC))
    m_useSystemTheme = Preferences::instance()->useSystemIconTheme();
    if (!m_useSystemTheme)
        QIcon::setThemeName("qbt-theme");
#else
    QIcon::setThemeName("qbt-theme");
#endif
```

New Code: 
```cpp
#if (defined(Q_OS_UNIX) && !defined(Q_OS_MAC))
    m_useSystemTheme = Preferences::instance()->useSystemIconTheme();
    if (!m_useSystemTheme)
#endif
        QIcon::setThemeName("qbt-theme");
```

My Proposed Code:  
```cpp
#if (defined(Q_OS_UNIX) && !defined(Q_OS_MAC))
    m_useSystemTheme = Preferences::instance()->useSystemIconTheme();
#endif
        QIcon::setThemeName("qbt-theme");
```

Because *within the diff*, the second `if` statement *appears* to have no effect (at least to me).

Reply:
>I admit that there is probably some nuance I am missing here…

The only nuance is knowledge of the C/C++ programming language.

```c++
QIcon::setThemeName("qbt-theme");
```
The line above is presented unconditionally so we will get on linux the following:
```c++
m_useSystemTheme = Preferences::instance()->useSystemIconTheme();
if (!m_useSystemTheme)
    QIcon::setThemeName("qbt-theme");
```

Reply:
>@Chocobo1 that's the way it was done before the "fixip!" commit. It's just @LordNyriox that doesn't get it.

Hmm, I didn't follow closely... however personally I won't write it that way.
BTW there is an redundant empty line at line 94.